net/net.go

```diff
@@ -1051,21 +1096,6 @@ func (n *net) loadRecords(
 			return nil, head, err
 		}
 
-		if validate {
-			dbody, err := event.GetBody(ctx, n, readKey)
-			if err != nil {
-				return nil, head, err
-			}
-
-			if err = identity.UnmarshalBinary(r.PubKey()); err != nil {
-				return nil, head, err
-			}
-
-			if err = connector.ValidateNetRecordBody(ctx, dbody, identity); err != nil {
```

Comment:
@dgtony curious what you think about this problem and solution. We can't validate the record here because it might rely on older DB instances. E.g., if we have these events in the chain: `create` <- `save`, `save` write validation requires the instance to already be created.

Reply:
As far as I understand the problem it may arise if your validation requires access to the previous state of the thread. I didn't actually realize that record validation can be "impure", i.e. depend on some information external to the record itself.
In this case there is no option besides validating along with applying. Failed record autoremoving, as Merlin pointed out isn't very clear, but imho it's fine. Record in the blockstore is in fact a part of internal state and may affect processing (e.g. chain loading), so user shouldn't directly manipulate it while system rollbacks to the clean state.